service/utils/channel_utils.js

```diff
@@ -98,7 +98,8 @@ function isDirectChannelVisible(userId, myPreferences, channel) {
 
 function isFavoriteChannel(myPreferences, channel) {
     const fav = myPreferences[`${Constants.CATEGORY_FAVORITE_CHANNEL}--${channel.id}`];
-    return fav && fav.value === 'true';
+    channel.isFavorite = fav && fav.value === 'true';
```

Comment:
Why are we mutating the channel here? If we need it anywhere else, that place should call this function directly